src/main/java/org/dcsa/tnt/service/impl/NotificationSignatureHandler.java

```diff
@@ -0,0 +1,374 @@
+                                            break;
+                                        case 401:
+                                            statusMessage = "Subscriber rejected the payload due to missing / invalid Authentication (HTTP 401).  Might need a Secret reset";
+                                            break;
+                                        case 403:
+                                            statusMessage = "Subscriber rejected the payload (HTTP 403)";
+                                            break;
+                                        default:
+                                            if (responseCode < 400) {
+                                                statusMessage = "Subscriber responded with cache code or an unknown redirection code (HTTP " + responseCode + ")!?";
+                                            } else if (responseCode < 500) {
+                                                statusMessage = "Subscriber rejected message suggesting a client error (HTTP " + responseCode + ")";
+                                            } else {
+                                                statusMessage = "Subscriber was unavailable (HTTP " + responseCode + ")";
+                                            }
+                                            break;
+                                    }
+                                    if (!respectRetryAfter || nextAttempt == null) {
+                                        nextAttempt = computeNextDelay(eventSubscriptionState);
+                                    }
+                                    eventSubscriptionState.setRetryAfter(nextAttempt);
+                                    log.debug("Notification for " + callbackUrl + " failed, it replied with: "
+                                            + responseCode + ": Will retry after " + nextAttempt + " (status: "
+                                            + statusMessage + ")");
+                                }
+                                eventSubscriptionState.setLastStatusMessage(statusMessage);
+                                return Mono.just(eventSubscriptionState);
+                            });
+                });
+    }
+
+    private OffsetDateTime parseRetryAfterHeader(String callbackUrl, String retryAfterRaw) {
+        OffsetDateTime parsedDateTime = null;
+
+        if (!retryAfterRaw.isEmpty() && Character.isDigit(retryAfterRaw.charAt(0))) {
+            try {
+                long delay = Long.parseLong(retryAfterRaw);
+                parsedDateTime = OffsetDateTime.now().plusSeconds(delay);
+            } catch (NumberFormatException e) {
+                log.debug(callbackUrl + " returned invalid Retry-After header (delay-variant); ignoring");
+            }
+        } else {
+            try {
+                parsedDateTime = OffsetDateTime.parse(retryAfterRaw, DateTimeFormatter.RFC_1123_DATE_TIME);

```

Comment:
Is this the "official" way to define 429 Retry after if it is a date?

Reply:
After checking - I can see that this is in fact so...

Reply:
I believe so. I used https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After as a reference for this parsing.